projects/docs/docs/manifests/project.md

```diff
@@ -814,3 +815,14 @@ Tuist automatically performs several template transformations for you
 :::note ExpressibleByStringInterpolation
 File header template can be initialized with a plain string, it is equivalent to using `.string(...)`
 :::
+
+### FileCodeGen
+
+The `FileCodeGen` enum represents the possible source file attributes for soruce code generation
```

Comment:
Worth a note that this is only applicable to a subset of file types (e.g. intent definitions and ML models)